src/common/UserAvatarWithPresence.js

```diff
@@ -41,3 +51,33 @@ export default class UserAvatarWithPresence extends PureComponent<Props> {
     );
   }
 }
+
+// Export the class with a tighter constraint on acceptable props, namely
+// that the type is an exact object type as usual.
+export default (UserAvatarWithPresence: ComponentType<Props>);
+
+/**
+ * A user avatar with a PresenceStatusIndicator in the corner.
+ *
+ * Use this in preference to the default export `UserAvatarWithPresence`.
+ * We're migrating from that one to this in order to avoid using emails.
+ *
+ * @prop [userId]
+ * @prop [size]
+ * @prop [onPress]
+ */
+export const UserAvatarWithPresenceById = connect<{| avatarUrl: AvatarURL, email: string |}, _, _>(
+  (state, props) => {
+    const user = getUserForId(state, props.userId);
+    return { avatarUrl: user.avatar_url, email: user.email };
+  },
+)(
+  // The type inference embedded in our `connect` type relies on seeing the
+  // exact Props type for the underlying component, complete with all the
+  // expected props.  Normally that's just how our Props types are in the
+  // first place, but here we have to provide it explicitly.
+  /* eslint-disable flowtype/generic-spacing */
+  (UserAvatarWithPresence: ComponentType<
```

Comment:
I think using react-redux's `connect` here is fine and convenient because we want to create a separate thing with a new interface, instead of modifying the original thing.

But when we settle on using this one and removing the old one, I wonder about trying out React Hooks, in particular, react-redux's `useReducer` (which is already available with the react-redux version we're on). True, we've gotten a pretty good handle on treating the "inner" and "outer" props in most places, with `SelectorProps`—but we won't have to do even that, when there's no wrapping component involved. 🙂

Reply:
Sure, that'd be interesting!

(`useSelector`, I think.)

Reply:
> (`useSelector`, I think.)

Ah of course 🤪.